### PR TITLE
Introduce StorageVersion

### LINKED ISSUE #833: Storage versioning and migration
### Problem to Solve

We need to create `STORAGE_VERSION` constant in `pallet-gear` to be able to perform runtime upgrades with migration.

### Possible Solution

Here is a useful list of successful runtime migration examples https://github.com/apopiak/substrate-migrations.

### Notes

_No response_

### PR DESCRIPTION
**Pull request by ukint-vs**
Resolves #833.

- add `StorageVersion` in pallets.
- add `migration.rs` templates for future migrations.

@gear-tech/dev

### CODE REVIEW

File: pallets/gear-program/src/migration.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Config, Pallet, Weight};
+
+/// Wrapper for all migrations of this pallet, based on `StorageVersion`.
+pub fn migrate<T: Config>() -> Weight {
+    use frame_support::traits::StorageVersion;
```
Comment (shamilsan) [inline pallets/gear-program/src/migration.rs:23]: Why this `use` is inside the function?

File: pallets/gear-program/src/lib.rs [hunk 2]
```diff
@@ -39,11 +42,15 @@ mod mock;
...
 mod tests;
 
+pub mod migration;
 pub mod weights;
 
 #[cfg(feature = "runtime-benchmarks")]
 mod benchmarking;
 
+/// The current storage version.
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```
Comment (shamilsan) [inline pallets/gear-program/src/lib.rs:52]: Isn't it better to have a different name for this constant in every pallet to not be confused when incrementing?
```suggestion
const PROGRAM_STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
```

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -127,7 +127,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
     // The version of the runtime specification. A full node will not attempt to use its native
     //   runtime in substitute for the on-chain Wasm runtime unless all of `spec_name`,
     //   `spec_version`, and `authoring_version` are the same between Wasm and native.
-    spec_version: 660,
+    spec_version: 680,
```
Comment (shamilsan) [inline runtime/src/lib.rs:130]: Increment with a reserve? 😄

File: pallets/gear-program/src/migration.rs (hunk 1 above)
Comment (ukint-vs) [reply pallets/gear-program/src/migration.rs:23]: Because `StorageVersion` needed only in this fn. Looks neat.

**Review by gshep — APPROVED** [review]

**Pull request merged**